protocol_rfcs/catalog-owned.md

```diff
@@ -0,0 +1,489 @@
+   the `cutOffCheckpoint` checkpoint. Also delete all the [log compaction files](#log-compaction-files)
+   having startVersion <= `cutOffCheckpoint`'s version.
+    - <ins>Also delete all the [staged commit files](#staged-commit) having version <=
+      `cutOffCheckpoint`'s version from the `_delta_log/_staged_commits` directory.</ins>
+
+--------
+
+> ***The next set of sections will be added to the existing spec just before [Iceberg Compatibility V1](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#iceberg-compatibility-v1) section***
+
+# Catalog-owned tables
+
+With this feature enabled, the [catalog](#terminology-catalogs) that manages the table becomes the
+source of truth for whether a given commit attempt succeeded.
+
+The table feature defines the parts of the [commit protocol](#commit-protocol) that directly impact
+the Delta table (e.g. atomicity requirements, publishing, etc). The Delta client and catalog
+together are responsible for implementing the Delta-specific aspects of commit as defined by this
+spec, but are otherwise free to define their own APIs and protocols for communication with each
+other.
+
+**NOTE**: Filesytem-based access to catalog-owned tables is not supported. Delta clients are
+expected to discover and access catalog-owned tables through the owning catalog, not by direct
+listing in the filesystem. This feature is primarily designed to warn filesystem-based readers that
+might attempt to access a catalog-owned table's storage location without going through the catalog
+first, and to block filesystem-based writers who could otherwise corrupt both the table and the
+catalog by failing to commit through the catalog.
+
+Before we can go into details of this protocol feature, we must first align our terminology.
+
+## Terminology: Commits
+
+A commit is a set of [actions](#actions) that transform a Delta table from version `v - 1` to `v`.
+A commit contains the same content as a [Delta file](#delta-log-entries).
+
+There are several types of commits:
+
+1. **Proposed commit**:  A commit that a Delta client has proposed for the next version of the
+   table. It could be "staged" or "inline". It will either become "ratified" or be rejected.
+
+2. **Staged commit** <a name="staged-commit"></a>: A proposed commit that is written to disk at
+   `_delta_log/_staged_commits/<v>.<uuid>.json`.
+    - Here, the `uuid` is a random UUID that is generated for each commit and `v` is the version
+      which is being committed, zero-padded to 20 digits.
+    - The catalog tracks only the location, not the content, of a staged commit.
+    - The mere existence of one of these files does not mean that the file is a _ratified_ commit.
+      It might correspond to a failed or in-progress commit attempt.
+    - The catalog is the source of truth around which staged commits are "ratified".
+
+3. **Inline commit** <a name="inline-commit"></a>: A proposed commit that is not written to disk but
```

Comment:
Inline commits can also be ratified, right?

Reply:
Yes. I think I need to make it clearer that both staged commits and inline commits are (1) proposed and then (2) ratified.